src/formatDate.js

```diff
@@ -8,7 +8,25 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateComponents = date.split(fromFormat[3]);
+  const result = [];
+
+  result[toFormat.indexOf('DD')] = dateComponents[fromFormat.indexOf('DD')];
+  result[toFormat.indexOf('MM')] = dateComponents[fromFormat.indexOf('MM')];
+
+  let year = fromFormat.includes('YYYY')
+    ? dateComponents[fromFormat.indexOf('YYYY')]
+    : dateComponents[fromFormat.indexOf('YY')];
+
+  if (!toFormat.includes('YYYY')) {
```

Comment:
The conditional check for the presence of 'YYYY' in the `toFormat` is correct, and the subsequent logic to slice the year if necessary is appropriate.